src/main/java/mate/academy/spring/service/validator/EmailValidator.java

```diff
@@ -0,0 +1,23 @@
+package mate.academy.spring.service.validator;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class EmailValidator implements ConstraintValidator<Email, String> {
+    private static final String EMAIL_PATTERN = "^[a-zA-Z0-9.!#$%&'*+/=?^_'{|}~-]"
+            + "+@((\\[[0-9]{1,3}\\.[0-9]{1,3}.\\[[0-9]{1,3}."
+            + "\\[[0-9]{1,3}])|(([a-zA-Z\\-0-9]+\\.)+[a-zA-Z]{2,}))$";
+
+    @Override
+    public boolean isValid(String email, ConstraintValidatorContext constraintValidatorContext) {
+        if (email == null) {
+            return false;
+        } else {
+            Pattern pattern = Pattern.compile(EMAIL_PATTERN);
```

Comment:
Let's create a constant of a type Pattern)

Reply:
Agree. There is no need to compile the same string patters each time you call this method